From 032457bd6e569f538652775635ca91981fecdf4b Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa@velocitylimitless.com>
Date: Mon, 29 Dec 2025 12:06:53 -0500
Subject: [PATCH 1/2] cmake: Correctly use path variables and install systemd
 data correctly

This ensures that this can be built and installed properly on various
Linux distributions (like e.g. Fedora Linux) leveraging standard
packaging conventions.
---
 CMakeLists.txt                                | 39 ++++++++++++++-----
 ...e => dynamic-hibernate-cleanup.service.in} |  2 +-
 ... => dynamic-hibernate-notifier.service.in} |  2 +-
 ...e => dynamic-hibernate-prepare.service.in} |  2 +-
 4 files changed, 32 insertions(+), 13 deletions(-)
 rename src/systemd/{dynamic-hibernate-cleanup.service => dynamic-hibernate-cleanup.service.in} (88%)
 rename src/systemd/{dynamic-hibernate-notifier.service => dynamic-hibernate-notifier.service.in} (87%)
 rename src/systemd/{dynamic-hibernate-prepare.service => dynamic-hibernate-prepare.service.in} (85%)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 93bdfcd..752b626 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0-only OR GPL-3.0-only OR LicenseRef-KDE-Accepted-GPL
 # SPDX-FileCopyrightText: 2025 Thomas Duckworth <tduck@filotimoproject.org>
+# SPDX-FileCopyrightText: 2025 Velocity Limitless, LLC
 
 cmake_minimum_required(VERSION 3.28)
 
@@ -18,39 +19,57 @@ add_custom_target(dynamic-hibernate
 
 install(
     PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/debug/dynamic-hibernate
-    DESTINATION ${KDE_INSTALL_LIBDIR}
+    DESTINATION ${KDE_INSTALL_LIBEXECDIR}
 )
 
 install(
     PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/debug/dynamic-hibernate-notifier
-    DESTINATION ${KDE_INSTALL_LIBDIR}
+    DESTINATION ${KDE_INSTALL_LIBEXECDIR}
+)
+
+configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/systemd/dynamic-hibernate-prepare.service.in
+    ${CMAKE_CURRENT_BINARY_DIR}/systemd/dynamic-hibernate-prepare.service
+    @ONLY
+)
+
+configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/systemd/dynamic-hibernate-cleanup.service.in
+    ${CMAKE_CURRENT_BINARY_DIR}/systemd/dynamic-hibernate-cleanup.service
+    @ONLY
+)
+
+configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/systemd/dynamic-hibernate-notifier.service.in
+    ${CMAKE_CURRENT_BINARY_DIR}/systemd/dynamic-hibernate-notifier.service
+    @ONLY
 )
 
 install(
     FILES
-        ${CMAKE_CURRENT_SOURCE_DIR}/src/systemd/dynamic-hibernate-prepare.service
-        ${CMAKE_CURRENT_SOURCE_DIR}/src/systemd/dynamic-hibernate-cleanup.service
-    DESTINATION ${KDE_INSTALL_LIBDIR}/systemd/system
+        ${CMAKE_CURRENT_BINARY_DIR}/systemd/dynamic-hibernate-prepare.service
+        ${CMAKE_CURRENT_BINARY_DIR}/systemd/dynamic-hibernate-cleanup.service
+    DESTINATION ${KDE_INSTALL_SYSTEMDUNITDIR}/system
 )
 
 install(
     FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/systemd/00-dynamic-hibernate-sleep.conf
-    DESTINATION ${KDE_INSTALL_LIBDIR}/systemd/sleep.conf.d
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/systemd/sleep.conf.d
 )
 
 install(
     FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/systemd/00-dynamic-hibernate-logind.conf
-    DESTINATION ${KDE_INSTALL_LIBDIR}/systemd/system/systemd-logind.service.d
+    DESTINATION ${KDE_INSTALL_SYSTEMDUNITDIR}/system/systemd-logind.service.d
 )
 
 install(
-    FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/systemd/dynamic-hibernate-notifier.service
-    DESTINATION ${KDE_INSTALL_LIBDIR}/systemd/user
+    FILES ${CMAKE_CURRENT_BINARY_DIR}/systemd/dynamic-hibernate-notifier.service
+    DESTINATION ${KDE_INSTALL_SYSTEMDUSERUNITDIR}
 )
 
 install(
     FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/notifier_dbus/org.dynamic_hibernate.DynamicHibernate.conf
-    DESTINATION ${KDE_INSTALL_FULL_DATAROOTDIR}/dbus-1/system.d
+    DESTINATION ${KDE_INSTALL_DBUSDIR}/system.d
 )
 
 ki18n_install(po)
diff --git a/src/systemd/dynamic-hibernate-cleanup.service b/src/systemd/dynamic-hibernate-cleanup.service.in
similarity index 88%
rename from src/systemd/dynamic-hibernate-cleanup.service
rename to src/systemd/dynamic-hibernate-cleanup.service.in
index d85260a..a84f477 100644
--- a/src/systemd/dynamic-hibernate-cleanup.service
+++ b/src/systemd/dynamic-hibernate-cleanup.service.in
@@ -12,7 +12,7 @@ ConditionPathExists=/var/dynamic-hibernate/dynamic-hibernate.swapfile
 
 [Service]
 Type=oneshot
-ExecStart=/usr/lib/dynamic-hibernate cleanup
+ExecStart=@KDE_INSTALL_FULL_LIBEXECDIR@/dynamic-hibernate cleanup
 
 [Install]
 WantedBy=systemd-hibernate-clear.service
diff --git a/src/systemd/dynamic-hibernate-notifier.service b/src/systemd/dynamic-hibernate-notifier.service.in
similarity index 87%
rename from src/systemd/dynamic-hibernate-notifier.service
rename to src/systemd/dynamic-hibernate-notifier.service.in
index a00ee20..9ba95fe 100644
--- a/src/systemd/dynamic-hibernate-notifier.service
+++ b/src/systemd/dynamic-hibernate-notifier.service.in
@@ -10,7 +10,7 @@ PartOf=graphical-session.target
 
 [Service]
 Type=simple
-ExecStart=/usr/lib/dynamic-hibernate-notifier
+ExecStart=@KDE_INSTALL_FULL_LIBEXECDIR@/dynamic-hibernate-notifier
 Restart=on-failure
 
 [Install]
diff --git a/src/systemd/dynamic-hibernate-prepare.service b/src/systemd/dynamic-hibernate-prepare.service.in
similarity index 85%
rename from src/systemd/dynamic-hibernate-prepare.service
rename to src/systemd/dynamic-hibernate-prepare.service.in
index f37ea6f..cadc033 100644
--- a/src/systemd/dynamic-hibernate-prepare.service
+++ b/src/systemd/dynamic-hibernate-prepare.service.in
@@ -10,7 +10,7 @@ Before=hibernate.target
 
 [Service]
 Type=oneshot
-ExecStart=/usr/lib/dynamic-hibernate create
+ExecStart=@KDE_INSTALL_FULL_LIBEXECDIR@/dynamic-hibernate create
 
 [Install]
 WantedBy=hibernate.target
-- 
GitLab


From 9011a67016ce924740e09cd6103e93636cebbea1 Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa@velocitylimitless.com>
Date: Mon, 29 Dec 2025 12:07:00 -0500
Subject: [PATCH 2/2] cmake: Allow setting the cargo build profile

This lets custom build profiles (like Fedora's "rpm" profile) work.
---
 CMakeLists.txt | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 752b626..fa6d83f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,18 +12,20 @@ include(KDEInstallDirs)
 include(ECMUninstallTarget)
 find_package(KF6I18n REQUIRED)
 
+set(CARGO_BUILD_PROFILE "release" CACHE STRING "Cargo build profile to use")
+
 add_custom_target(dynamic-hibernate
     ALL
-    COMMAND cargo build --target-dir ${CMAKE_CURRENT_BINARY_DIR}
+    COMMAND cargo build --profile=${CARGO_BUILD_PROFILE} --target-dir ${CMAKE_CURRENT_BINARY_DIR}
 )
 
 install(
-    PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/debug/dynamic-hibernate
+    PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/${CARGO_BUILD_PROFILE}/dynamic-hibernate
     DESTINATION ${KDE_INSTALL_LIBEXECDIR}
 )
 
 install(
-    PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/debug/dynamic-hibernate-notifier
+    PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/${CARGO_BUILD_PROFILE}/dynamic-hibernate-notifier
     DESTINATION ${KDE_INSTALL_LIBEXECDIR}
 )
 
-- 
GitLab

